app/database/migrations/2018_07_13_212855_add_soft_deletes_user_answer_table.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddSoftDeletesUserAnswerTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('user_answer', function (Blueprint $table) {
+            $table->softDeletes();
```

Comment:
What is this for?  Answers aren't saved in the database until the user clicks "Submit".  After the answers are submitted there's no reason to ever delete them.

Reply:
Yes this is soft deleting the users old answer so that we don't have to query for the lastest one. If user is sending the answer for the same location again then soft delete the previous one and save the new one.

Reply:
This migration script will add `deleted_at` field in the user_answer table.

Reply:
@joshi1983 It's fixed now!

Reply:
I like the idea as it speeds up the calculation.  It would be acting like a cache of the expression "is this the latest answer from a user on this location?" similar to how ratings_cache maintains other information that make the rating calculation faster.  Both have the drawbacks associated with maintaining duplicated or calculable data that could become inconsistent but the performance improvement would outweigh that drawback.

This migration would be sufficient if we were always recreating the database from scratch but this also needs to migrate old user_answer data in production.  A database containing multiple user_answer records on the same location and user will not have everything but the latest answers indicated as soft-deleted after running this script.  In other words, this deleted_at field starts its existence inconsistent with the expression "is this the latest answer from a user on this location?".

After adding the field in the up method, initialize the field to correctly mark old user_answer records as soft-deleted.

Reply:
I'll handle getting the migration script adjusted so it properly initializes the new deleted_at field.